hypothesis-python/src/hypothesis/internal/conjecture/choicetree.py

```diff
@@ -70,13 +78,7 @@ def finish(self):
         self.__finished = True
         assert len(self.__node_trail) == len(self.__choices) + 1
 
-        next_value = list(self.__choices)
-        while next_value:
-            next_value[-1] -= 1
-            if next_value[-1] < 0:
-                next_value.pop()
-            else:
-                break
```

Comment:
I notice that we no longer explicitly compute the next prefix.

Is that because we can rely on it getting found automatically when the choice loop rejects one of the last-prefix values for being exhausted?

Reply:
Yes, exactly. I realised we were essentially duplicating that logic here so I removed it. Sorry I should have made mention of this in the PR comment.